packages/suite/src/views/wallet/coinmarket/common/CoinmarketForm/CoinmarketFormInput/CoinmarketFormInputCryptoSelect.tsx

```diff
@@ -56,40 +57,57 @@ export const CoinmarketFormInputCryptoSelect = <
         [buildCryptoOptions, supportedCryptoCurrencies, sendCryptoSelectValue],
     );
 
+    const handleSelectChange = (selectedCryptoId: string) => {
+        const findOption = options.find(
+            option => option.type === 'currency' && option.value === selectedCryptoId,
+        ) as CoinmarketCryptoSelectItemProps | undefined;
+
+        if (!findOption) return;
+
+        if (isCoinmarketExchangeOffers(context)) {
+            context.setValue(FORM_RECEIVE_CRYPTO_CURRENCY_SELECT, findOption);
+        } else {
+            context.setValue(FORM_CRYPTO_CURRENCY_SELECT, findOption);
+        }
+
+        setIsModalActive(false);
+    };
+
+    const getNetworks = () => {
+        const eth = getNetwork('eth');
+        const polygon = getNetwork('matic');
+        const sol = getNetwork('sol');
+        const bnb = getNetwork('bnb');
+
+        const getParams = (network: Network) => ({
+            name: network.name,
+            symbol: network.symbol,
+            coingeckoId: network.coingeckoId ?? network.symbol,
+            coingeckoNativeId: network.coingeckoNativeId,
+        });
+
+        return [getParams(eth), getParams(polygon), getParams(sol), getParams(bnb)];
+    };
+
     return (
         <>
             <CoinmarketFormInputLabel label={label} />
+            {isModalActive && (
+                <SelectAssetModal
```

Comment:
On network click it would be nice to deselect it

Reply:
Updated